.github/workflows/main.yml

```diff
@@ -4,27 +4,49 @@ on: [push, pull_request]
 
 jobs:
   Build:
-    name: ${{ matrix.platform.name }}
-    runs-on: ${{ matrix.platform.os }}
+    name: '${{ matrix.platform.name }}'
+    runs-on: '${{ matrix.platform.os }}'
     strategy:
       fail-fast: false
       matrix:
         platform:
-        - { name: Windows, os: windows-latest, msvc: true }
-        - { name: Linux,   os: ubuntu-latest }
-        - { name: MacOS,   os: macos-latest }
+        - { name: 'Linux',                    os: ubuntu-latest,  shell: sh }
+        - { name: 'MacOS',                    os: macos-latest,   shell: sh }
+        - { name: 'Windows msys2 (mingw32)',  os: windows-latest, shell: 'msys2 {0}', msystem: mingw32, msys-env: mingw-w64-i686 }
+        - { name: 'Windows msys2 (mingw64)',  os: windows-latest, shell: 'msys2 {0}', msystem: mingw64, msys-env: mingw-w64-x86_64 }
+        - { name: 'Windows msys2 (clang32)',  os: windows-latest, shell: 'msys2 {0}', msystem: clang32, msys-env: mingw-w64-clang-i686 }
+        - { name: 'Windows msys2 (clang64)',  os: windows-latest, shell: 'msys2 {0}', msystem: clang64, msys-env: mingw-w64-clang-x86_64 }
+        - { name: 'Windows msys2 (ucrt64)',   os: windows-latest, shell: 'msys2 {0}', msystem: ucrt64,  msys-env: mingw-w64-ucrt-x86_64 }
```

Comment:
Do we really need the ucrt version? We already build with no libc/crt dependency when targeting windows.

Reply:
msys2 advises to use ucrt64 (https://www.msys2.org/docs/environments/)

Reply:
But feel free to prune combinations. We don't want ci to become excessive.
Or select a few for ci, and run everything in a cron job.

Reply:
My suggestion is dropping ucrt here.

Reply:
I'll remove it once everything's in the green.